src/metric/elementwise_metric.cu

```diff
@@ -304,13 +304,17 @@ struct EvalGammaNLogLik {
   }
 
   XGBOOST_DEVICE bst_float EvalRow(bst_float y, bst_float py) const {
-    const bst_float eps = 1e-16f;
-    if (y < eps) y = eps;
-    bst_float psi = 1.0;
+    py = std::max(py, 1e-6f);
+    // hardcoded dispersion.
+    float constexpr kPsi = 1.0;
```

Comment:
Do we have a plan to allow for a different value for dispersion in the future?

Reply:
The dispersion should be estimated from data.  We can put more efforts to clarity the choice or add estimation in the code,right now I'm just adding notes.

Reply:
It can be treated as a hyperparameter.

Reply:
I'm not sure yet,as you can see the metric doesn't use label at all.

Reply:
The dependency on labels is hidden in dispersion.

Reply:
Wait, why is `std::log(y)` equal to 1.0f?

Reply:
Where?

Reply:
See my next comment, which you just responded.